### PR TITLE
kafka: create separate TopicCreator type

### PR DESCRIPTION
**Pull request by axw**
This allows one to use a single Manager to create topics with differing partition counts and configs. It also enables one to create a Manager without specifying PartitionCount, e.g. in case one just wants to list topics.

### CODE REVIEW

File: kafka/topiccreator.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	responses, err := c.m.client.CreateTopics(
+		ctx,
+		int32(c.partitionCount),
+		-1, // default.replication.factor
+		c.topicConfigs,
+		topicNames...,
+	)
+	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "CreateTopics returned an error")
```
Comment (dmathieu) [inline kafka/topiccreator.go:111]: Why not report the actual error?

```suggestion
		span.SetStatus(codes.Error, err.Error())
```

File: kafka/topiccreator.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		createTopicParamsFields = append(createTopicParamsFields,
+			zap.Reflect("topic_configs", c.origTopicConfigs),
+		)
+	}
+	var createErrors []error
+	for _, response := range responses.Sorted() {
+		logger := c.m.cfg.Logger.With(zap.String("topic", response.Topic))
+		if err := response.Err; err != nil {
+			if errors.Is(err, kerr.TopicAlreadyExists) {
+				logger.Debug("kafka topic already exists")
```
Comment (dmathieu) [inline kafka/topiccreator.go:127]: How about adding an attribute to the span here, or an event?

File: kafka/topiccreator.go (hunk 1 above)
Comment (axw) [reply kafka/topiccreator.go]: SGTM. I don't think we do anything with the status message at the moment, but perhaps we should.

File: kafka/topiccreator.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		createTopicParamsFields = append(createTopicParamsFields,
+			zap.Reflect("topic_configs", c.origTopicConfigs),
+		)
+	}
+	var createErrors []error
+	for _, response := range responses.Sorted() {
+		logger := c.m.cfg.Logger.With(zap.String("topic", response.Topic))
+		if err := response.Err; err != nil {
+			if errors.Is(err, kerr.TopicAlreadyExists) {
+				span.AddEvent("kafka topic already exists", trace.WithAttributes(
```
Comment (dmathieu) [inline kafka/topiccreator.go:127]: I just remembered: our agent has no notion of events.
So right now, apmotel doesn't do anything with them. While doing this seems like the right thing to do, it's currently going into `/dev/null`.

We should maybe do both logging and eventing?

File: kafka/topiccreator.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+					semconv.MessagingDestinationKey.String(response.Topic),
+				))
+			} else {
+				span.RecordError(err)
+				span.SetStatus(codes.Error, "failed to create one or more topic")
```
Comment (dmathieu) [inline kafka/topiccreator.go:132]:
```suggestion
				span.SetStatus(codes.Error, err.Error())
```

File: kafka/topiccreator.go (hunk 3 above)
Comment (axw) [reply kafka/topiccreator.go:130]: Ah, good point. In theory we could translate span events to logs in the agent, but for now I have added the log message back.

File: kafka/topiccreator.go (hunk 4 above)
Comment (axw) [reply kafka/topiccreator.go]: Done.

File: kafka/topiccreator_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+			"retention.ms": "123",
+		},
+	})
+	require.NoError(t, err)
+
+	var createTopicsRequest *kmsg.CreateTopicsRequest
+	cluster.ControlKey(kmsg.CreateTopics.Int16(), func(req kmsg.Request) (kmsg.Response, error, bool) {
+		createTopicsRequest = req.(*kmsg.CreateTopicsRequest)
+		return &kmsg.CreateTopicsResponse{
+			Version: 7,
```
[the remark this answers is not in the document]
Comment (axw) [reply kafka/topiccreator_test.go:85]: Thank you :)

**Pull request merged**